src/main/resources/liquibase/crop_changelog/studies_TBD.xml

```diff
@@ -0,0 +1,626 @@
+        <sql dbms="mysql" splitStatements="true">
+                 UPDATE projectprop SET type_id = 1815 WHERE variable_id = 8230 AND type_id = 1804;
+        </sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="studies-TBD-16">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="stock" columnName="cross_value"/>
+			</not>
+		</preConditions>
+		<comment>Add 'cross_value' column to 'stock' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			ALTER TABLE STOCK ADD `cross_value` VARCHAR(5000) DEFAULT '-' NOT NULL;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="studies-TBD-17">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<columnExists tableName="stock" columnName="cross_value"/>
+				<sqlCheck expectedResult="1">
+					SELECT count(1) > 0 FROM stockprop WHERE type_id = 8377;
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Migrate cross values from 'stockprop' table to 'cross_value' column in 'stock' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE stock s INNER JOIN stockprop sp ON s.stock_id = sp.stock_id
+			SET s.cross_value = sp.value
+			WHERE sp.type_id = 8377 AND sp.value IS NOT NULL AND sp.value != '-';
+
+			DELETE FROM stockprop WHERE type_id = 8377;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="studies-TBD-18">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<columnExists tableName="project" columnName="generation_level" />
+			</not>
+		</preConditions>
+		<comment>
+			Add 'generation_level' column: indicates preferred cross expansion (pedigree) level for the entire study
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			# a property to complement (or potentially migrate) the stored pedigree in stock.cross_value
+			alter table project
+				add generation_level int null comment 'cross expansion (pedigree) level';
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="studies-TBD-19">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                SELECT count(1) > 0 FROM projectprop WHERE variable_id = 8360
+            </sqlCheck>
+        </preConditions>
+        <comment>Replace the SEED_SOURCE variable by ENTRY_SOURCE IN projectprop AND stockprop</comment>
+        <sql dbms="mysql" splitStatements="true">
+			# Replace SEED_SOURCE variable by ENTRY_SOURCE
+			UPDATE projectprop SET variable_id = ( SELECT cvterm_id FROM CVTERM WHERE NAME ='ENTRY_SOURCE' ), ALIAS = 'ENTRY_SOURCE', TYPE_ID = 1815 WHERE variable_id = 8360;
+			# Update cvterm_id 8360 from (SEED_SOURCE) to ENTRY_SOURCE cvterm_id
+			UPDATE stockprop SET TYPE_ID = ( SELECT cvterm_id FROM CVTERM WHERE NAME ='ENTRY_SOURCE' )
+			                 WHERE TYPE_ID = 8360;
+        </sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="studies-TBD-20">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
```

Comment:
@cuenyad you dont need to use cv_id, cvterm_id should be enough for the precondition